src/lib/components/LeafletMap/README.md

```diff
@@ -0,0 +1,940 @@
+```javascript
+{
+    ...
+    "shader": {
+        "type": "soft-hillshading",
+        "elevationScale": 0.03, // Optional
+    },
+    ...
+}
+```
+ 
+<br>
+ 
+##### Soft hillshading options
+| Name           | Type          | Description                                                                                                       | Default   |
+|----------------|---------------|-------------------------------------------------------------------------------------------------------------------|-----------|
+| elevationScale | Number        | A variable for scaling the generated elevation of the image. The greater the scale, the higher the "_mountains_". | 0.03      |
+| lightDirection | Array<Number> | A vector of length 3 indicating the direction from the surface to the sun.                                        | [1, 1, 1] |
+ 
+<br>
+ 
+-------------------------------
+ 
+<br>
+ 
+## 🎙 Listeners
+ 
+There are some listeners the python user can access using callbacks, such as the coordinates of a mouse click or a drawing.
+ 
+Example:
+ 
+```python
+app.layout = html.Div(
+    children=[
+        leaflet_map_component,
+        html.Pre(id="polygon")
+    ]
+)
+ 
+@app.callback(
+     Output("polygon", "children"),
+     [Input("example-map", "polygon_points")]
+)
+def get_polygon_coords(coords):
+    return f"polygon coordinates: {json.dumps(coords)}"
+```
+ 
+<br>
+
+| Name                | Output format             | Description                                       |
+|---------------------|---------------------------|---------------------------------------------------|
+| **click_position**  | `{[x, y]}`                  | Coordinates of the last clicked area of the map   |
+| **marker_point**    | `{[x, y]}`                  | Coordinates of the last placed marker on the map  |
+| **polyline_points** | `{[[x1, y1], ..., [xn, yn]]}` | Coordinates of the last drawn polyline on the map |
+| **polygon_points**  | `{[[x1, y1], ..., [xn, yn]]}` | Coordinates of the last drawn polygon on the map  |
+| **clicked_shape**  | [Shape object](#shapes-and-drawings) | JSON object containing the details of last shape clicked   |
+
+<br>
+
+--------------
+
+<br>
+
+## Serving tiles with dash
+
+Dash is using an underlying flask server during runtime which can be accessed from the dash app. We can use this flask server to serve tiles.
+
+Assuming you have pregenerated tiles ready to go on a disk, this example of a very simple tile server will work.
+
+```python
+import os.path
+from flask import Flask, send_file
+
+class tile_server():
+
+    def __init__(self, flask_server):
+
+        app = flask_app
+
+        @app.route('/tiles/<zoom>/<y>/<x>')
+        def tiles(zoom, x, y):
+            default = r'_path_to_default_tile\tiles\0\{zoom}\{x}\{y}.png'
+            filename = r'_path_to_tiles\tiles\0\%s\%s\%s.png' % (zoom, x, y)
```

Comment:
Have not tested this, but does this work instead? https://www.python.org/dev/peps/pep-0498/#raw-f-strings

```suggestion
            filename = fr"_path_to_tiles\tiles\0\{zoom}\{x}\{y}.png"
```